nodeadm/internal/containerd/runtime_config.go

```diff
@@ -0,0 +1,78 @@
+package containerd
+
+import (
+	"slices"
+	"strings"
+
+	"github.com/awslabs/amazon-eks-ami/nodeadm/internal/api"
+	"github.com/awslabs/amazon-eks-ami/nodeadm/internal/util"
+	"github.com/pelletier/go-toml/v2"
+	"go.uber.org/zap"
+)
+
+type instanceTypeMixin struct {
+	instanceFamilies []string
+	apply            func(*[]byte) error
+}
+
+func (m *instanceTypeMixin) matches(cfg *api.NodeConfig) bool {
+	instanceFamily := strings.Split(cfg.Status.Instance.Type, ".")[0]
+	return slices.Contains(m.instanceFamilies, instanceFamily)
+}
+
+var (
+	// TODO: fetch this list dynamically
+	nvidiaInstances         = []string{"p3", "p3dn", "p4d", "p4de", "p5", "g4", "g4dn", "g5", "g6"}
```

Comment:
```suggestion
	nvidiaInstances         = []string{"p3", "p3dn", "p4d", "p4de", "p5", "g4", "g4dn", "g5", "g6", "g6e"}
```

Hopefully pulling this from the EC2 API is a fast follow 😅

Reply:
I would actually vote we add this now, or proactively add the instance types we know are launching fairly soon.

Reply:
I'd rather do it on a separate PR, this one is big enough. In the meantime any other instance type beside `g6e` that's worth adding?

Reply:
makes sense - no, just `g6e` is missing from the current list of supported instances. we should also have a "louder" failure for unsupported instances (i.e. - `p2`) if possible